### PR TITLE
Fix connect() and disconnect() behaviour (#226)

### PR DESCRIPTION
**Pull request by macbeth8c**
Fix connect() and disconnect() behaviour (#226)

### CODE REVIEW

File: lib/clients/websocket.js [hunk 1]
```diff
@@ -41,11 +41,10 @@ class WebsocketClient extends EventEmitter {
   }
 
   disconnect() {
-    if (!this.socket) {
+    if (!this.socket || this.socket.readyState !== Websocket.OPEN) {
```
Comment (fb55) [inline lib/clients/websocket.js]: Should also consider `CONNECTING`

File: lib/clients/websocket.js [hunk 2]
```diff
@@ -28,8 +28,8 @@ class WebsocketClient extends EventEmitter {
   }
 
   connect() {
-    if (this.socket) {
-      this.socket.close();
+    if (this.socket && this.socket.readyState !== Websocket.CLOSED) {
```
[the remark this answers is not in the document]
Comment (macbeth8c) [reply lib/clients/websocket.js]: The CLOSING was omitted on purpose. What is a meaning of connecting when the precious connection is not yet closed? Please explain

File: lib/clients/websocket.js (hunk 1 above)
Comment (macbeth8c) [reply lib/clients/websocket.js]: The CONNECTING was omitted on purpose. The meaning here might be that someone wants to abort the connection. In this case the socket will abort and throw an exception 'WebSocket was closed before the connection was established'. If this is the behaviour we want, I can fix it.

File: lib/clients/websocket.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (fb55) [reply lib/clients/websocket.js]: The socket isn't usable anymore if the state is `CLOSING`, so calling `connect` should be legitimate.

File: lib/clients/websocket.js (hunk 1 above)
Comment (fb55) [reply lib/clients/websocket.js]: Interesting. That error might still be more descriptive than the current message, but I'm fine with omitting it.

**Pull request merged**